### DISCUSSION TITLE
Does Unleash support any cache mechanisms?

### DISCUSSION THREAD

**Discussion by LittleHawk03** [Unleash/unleash#6710] [Q&A] [answered]
Does Unleash support any cache mechanisms such as redis or in-mem cache?

I ask because I'm not sure if every time edge or SDKs fetches data from unleash-server, it will impact the database to get the data or not.

**Comment by sighphyre** [comment, marked as answer]
Hey @LittleHawk03,

We actually have a whole bunch of caching going on internally in Unleash, Edge and the SDKs. Typically, a call to retrieve the features data made by Edge or an SDK will be read directly from memory and the response sent from that. That memory blob gets updated in the background so there's no direct correlation between db impact and calls from SDKs/Edge.

Pressure on the database from retrieving features typically shouldn't be a problem, unless you have upwards of tens of thousands of toggles and you're not using Edge to control the flow of data. Having thousands of toggles is generally not ideal, since it'll be tough to understand how your toggles are actually affecting your live system so in a typical setup this isn't a problem.

Edge and SDKs also respect ETags, so usually it's a read from memory and an empty 304 response sent back.

So TLDR: you probably don't need Redis here

**Reply by LittleHawk03** [reply]
Thank you so much for the reply. =)))

**Reply by sighphyre** [reply]
Nothing but pleasure!
